src/server/services/wfs/qgswfsutils.cpp

```diff
@@ -153,6 +153,35 @@ namespace QgsWfs
       request.setFlags( QgsFeatureRequest::ExactIntersect | QgsFeatureRequest::NoFlags );
       return request;
     }
+    else if ( filterElem.firstChildElement().tagName() == QLatin1String( "And" ) &&
+              !filterElem.firstChildElement().firstChildElement( QLatin1String( "BBOX" ) ).isNull() )
+    {
+      QDomElement childElem = filterElem.firstChildElement().firstChildElement();
+      while ( !childElem.isNull() )
+      {
+        QDomElement childFilterElement = filterElem.ownerDocument().createElement( QLatin1String( "Filter" ) );
+        childFilterElement.appendChild( childElem.cloneNode( true ) );
+        QgsFeatureRequest childRequest = parseFilterElement( typeName, childFilterElement );
+        if ( childElem.tagName() == QLatin1String( "BBOX" ) )
+        {
+          if ( request.filterRect().isEmpty() )
+          {
+            request.setFilterRect( childRequest.filterRect() );
+          }
+          else
+          {
+            request.setFilterRect( request.filterRect().intersect( &childRequest.filterRect() ) );
+          }
+        }
+        else
+        {
+          request.setFilterExpression( childRequest.filterExpression()->expression() );
```

Comment:
Are there potentially more than two child elements and the filter expression is overwritten repeatedly? (disclaimer: I've got no idea and do not know this code)

Reply:
Exact, the standard says : 
```<xsd:choice minOccurs="2" maxOccurs="unbounded">```
Multiple expressions need to be wrapped in an "and" expression node.

Reply:
so the risk of multiple expressions colliding is a real risk?